experimental/InvariantTheory/invariant_rings.jl

```diff
@@ -423,3 +359,107 @@ function heisenberg_group(n::Int)
   end
   return MatrixGroup(n, K, [ M1, M2 ])
 end
+
+################################################################################
+#
+#  Molien series
+#
+################################################################################
+
+# Some functionality via Singular
+function __molien_series_via_singular(IR::InvRing{T}) where {T <: Union{FlintRationalField, AnticNumberField, Nemo.GaloisField, Nemo.GaloisFmpzField}}
+  return reynolds_molien_via_singular(IR)[2]
+end
+
+function _molien_series_via_singular(S::PolyRing, IR::InvRing{T}) where {T <: Union{FlintRationalField, AnticNumberField}}
+  mol = __molien_series_via_singular(IR)
+  # Singular does not build a new polynomial ring for the univariate Hilbert series
+  # (how could it after all), but uses the first variable of the given ring.
+
+  R = polynomial_ring(IR)
+  K = coefficient_ring(IR)
+  # Need an extra coercion here while waiting for https://github.com/Nemocas/AbstractAlgebra.jl/pull/1009
+  #return to_univariate(S, R(mol[1, 1]))//to_univariate(S, R(mol[1, 2]))
+  return S(to_univariate(S, R(mol[1, 1])))//S(to_univariate(S, R(mol[1, 2])))
+end
+
+function _molien_series_via_singular(S::PolyRing, IR::InvRing{T}) where {T <: Union{Nemo.GaloisField, Nemo.GaloisFmpzField}}
+  @assert !ismodular(IR)
+  mol = molien_series_via_singular(IR)
```

Comment:
Should this be this? If not, why not?
```suggestion
  mol = __molien_series_via_singular(IR)
```